src/Lib/Route/RouteScanner.php

```diff
@@ -27,12 +27,18 @@ class RouteScanner
      */
     private array $routes;
 
+    /**
+     * @var \SwaggerBake\Lib\Configuration
+     */
+    private Configuration $config;
+
     /**
      * @param \Cake\Routing\Router $router CakePHP Router
      * @param \SwaggerBake\Lib\Configuration $config Swagger Configuration
      */
     public function __construct(private Router $router, Configuration $config)
     {
+        $this->config = $config;
```

Comment:
Instead of storing the entire configuration object in a class property, consider instead creating a property `$hotReloadEnabled` and populating it's value from the config, like you can see was done with `$prefix` in the line below this one.

Reply:
The thing is that ModelScanner was already implemented this way, a config property is created with the whole configuration. I tried to follow the same way for code standardization purpose.